hypervisor/src/mshv/mod.rs

```diff
@@ -879,34 +902,40 @@ impl vm::Vm for MshvVm {
     }
 
     /// Creates a guest physical memory region.
-    fn create_user_memory_region(&self, user_memory_region: MemoryRegion) -> vm::Result<()> {
+    fn create_user_memory_region(
+        &self,
+        user_memory_region: hypervisor::MemoryRegion,
+    ) -> vm::Result<()> {
         // No matter read only or not we keep track the slots.
         // For readonly hypervisor can enable the dirty bits,
         // but a VM exit happens before setting the dirty bits
         self.dirty_log_slots.write().unwrap().insert(
-            user_memory_region.guest_pfn,
+            user_memory_region.guest_phys_addr,
             MshvDirtyLogSlot {
-                guest_pfn: user_memory_region.guest_pfn,
-                memory_size: user_memory_region.size,
+                guest_pfn: user_memory_region.guest_phys_addr >> PAGE_SHIFT,
+                memory_size: user_memory_region.memory_size,
             },
         );
 
         self.fd
-            .map_user_memory(user_memory_region)
+            .map_user_memory(user_memory_region.into())
             .map_err(|e| vm::HypervisorVmError::CreateUserMemory(e.into()))?;
         Ok(())
     }
 
     /// Removes a guest physical memory region.
-    fn remove_user_memory_region(&self, user_memory_region: MemoryRegion) -> vm::Result<()> {
+    fn remove_user_memory_region(
+        &self,
+        user_memory_region: hypervisor::MemoryRegion,
+    ) -> vm::Result<()> {
         // Remove the corresponding entry from "self.dirty_log_slots" if needed
         self.dirty_log_slots
             .write()
             .unwrap()
-            .remove(&user_memory_region.guest_pfn);
+            .remove(&(user_memory_region.guest_phys_addr >> PAGE_SHIFT));
```

Comment:
Avoiding having to manually translate fields is the whole point of the `From` and `Into` traits.

You can just shadow `user_memory_region` to be less error prone.


Something roughly like:
`let user_memory_region: mshv_memory_region = user_memory_region.into();`

The same pattern can be used in other places.

Reply:
This seems better than what I have done right now, I will do this for both mshv and kvm.

Reply:
Also, the make_user_memory_region is redundant, I can just implement MemoryRegion::new() for the same functionality?